pkg/filesystem/local_directory_windows.go

```diff
@@ -0,0 +1,1105 @@
+			for i := uint32(0); i < numOutStruct; i++ {
+				if allocRangeBufferPtr.FileOffset != lastOffset {
+					offset = lastOffset
+					break
+				}
+				lastOffset = allocRangeBufferPtr.FileOffset + allocRangeBufferPtr.Length
+				allocRangeBufferPtr = (*FILE_ALLOCATED_RANGE_BUFFER)(unsafe.Pointer(
+					uintptr(unsafe.Pointer(allocRangeBufferPtr)) + unsafe.Sizeof(bufferSize)))
+			}
+			offset = lastOffset
+		}
+	}
+	nextOffset, err := setFilePointer(handle, offset, windows.FILE_BEGIN)
+	return nextOffset, err
+}
+
+func (f localFileReadWriter) GetNextRegionOffset(offset int64, regionType RegionType) (int64, error) {
+	handle := windows.Handle(f.Fd())
+	var fileInfo windows.ByHandleFileInformation
+	err := windows.GetFileInformationByHandle(handle, &fileInfo)
+	if err != nil {
+		return 0, err
+	}
+	fileAttributes := fileInfo.FileAttributes
+	// only files that have either attribute have zero ranges known to the os
+	isSparse := (fileAttributes & windows.FILE_ATTRIBUTE_SPARSE_FILE) == windows.FILE_ATTRIBUTE_SPARSE_FILE
+	isCompressed := (fileAttributes & windows.FILE_ATTRIBUTE_COMPRESSED) == windows.FILE_ATTRIBUTE_COMPRESSED
+	fileSize := (uint64(fileInfo.FileSizeHigh) << 32) | uint64(fileInfo.FileSizeLow)
+	if offset >= int64(fileSize) {
+		return 0, io.EOF
+	}
+	if isSparse || isCompressed {
+		return getNextRegionOffsetSparse(handle, offset, regionType, fileSize)
+	}
+	var whence uint32
+	switch regionType {
+	case Data:
+		whence = windows.FILE_BEGIN
+	case Hole:
+		offset = 0
+		whence = windows.FILE_END
+	}
+	nextOffset, err := setFilePointer(handle, offset, whence)
+	return nextOffset, err
+}
+
+func (d *localDirectory) OpenRead(name path.Component) (FileReader, error) {
+	f, err := d.open(name, DontCreate, os.O_RDONLY)
+	if err != nil {
+		return nil, err
+	}
+	return localFileReadWriter{File: f}, nil
+}
+
+func (d *localDirectory) OpenReadWrite(name path.Component, creationMode CreationMode) (FileReadWriter, error) {
+	f, err := d.open(name, creationMode, os.O_RDWR)
+	if err != nil {
+		return nil, err
+	}
+	return localFileReadWriter{File: f}, nil
+}
+
+func (d *localDirectory) OpenWrite(name path.Component, creationMode CreationMode) (FileWriter, error) {
+	f, err := d.open(name, creationMode, os.O_WRONLY)
+	if err != nil {
+		return nil, err
+	}
+	return localFileReadWriter{File: f}, nil
+}
+
+func (d *localDirectory) Link(oldName path.Component, newDirectory Directory, newName path.Component) error {
+	defer runtime.KeepAlive(d)
+
+	return newDirectory.Apply(localDirectoryLink{
+		oldHandle: d.handle,
+		oldName:   oldName,
+		newName:   newName,
+	})
+}
+
+func (d *localDirectory) Clonefile(oldName path.Component, newDirectory Directory, newName path.Component) error {
+	return status.Error(codes.Unimplemented, "Clonefile is only supported on Darwin")
+}
+
+var executableExts = [...]string{".COM",".EXE",".BAT",".CMD",".VBS",".VBE",".JS",".JSE",".WSF",".WSH",".MSC",".CPL"}
+
+func isExecutable(fileName string) bool {
```

Comment:
Another option would be to always return true on Windows, right? I've noticed that Bazel itself already makes almost everything in input roots executable, regardless of permissions on the host (even on Linux). It wouldn't really harm us if we did the same thing on the remote side.